go/oauth2util/util.go

```diff
@@ -0,0 +1,75 @@
+package oauth2util
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+)
+
+// Default 3LO authorization handler. Prints the authorization URL on stdout
+// and reads the verification code from stdin.
+func defaultAuthorizeFlowHandler(authorizeUrl string) (string, error) {
+	// Print the url on console, let user authorize and paste the token back.
+	fmt.Printf("Go to the following link in your browser:\n\n   %s\n\n", authorizeUrl)
+	fmt.Println("Enter verification code: ")
+	var code string
+	fmt.Scanln(&code)
+	return code, nil
+}
+
+// NewClient create a new OAuth2 Client.
+//
+// key: a JSON string that represents either an OAuth client ID or a
+// service account key.
+// scope: A space separated scope codes per
+//     [OAuth 2.0 spec](https://tools.ietf.org/html/rfc6749).
+func NewClient(key []byte, scope []string, authorizeHandler func(string) (string, error)) (oauth2.TokenSource, error) {
```

Comment:
`type AuthorizeHandler func(string) (string, error)` would be good to have, then we can document what this handler is and its responsibilities very comprehensively.